### DISCUSSION TITLE
2022-01-05 Budget Subcomittee Meeting

### DISCUSSION THREAD

**Discussion by andy-blum** [ddev/ddev#3413] [DDEV Advisory Group]
# Minutes

See [Meeting Recording](https://www.dropbox.com/s/46wq2tg5hox33vn/2022-12-01%20DDEV%20Advisory%20Group.mp4?dl=0)

Agenda: Discuss a potential budget for DDEV.
* Start recording
* Ask somebody to take notes on the attached doc
* Randy’s thinking:

Big goals:
- financial support for development
  - Starting with goal of US$60K/year for one developer (Randy) (no benefits)
  - Stretch goal of second developer so all is not dependent on things Randy knows, for continuity into the future, and for projects that we can’t take on with just one dev (like a GUI).
- Smaller goals:
  -self-hosted test runner replacement - 4 machines a year? Right now we could use 4 mac Minis, 2 arm64 + 2 amd64 to replace aging laptops as old as 2013. Windows next year? Perhaps a budget of $2k/year
  - Misc expenses
    - Dockerhub: $300/year
    - Discord nitro (boost): $120/year
    - PhpStorm and Goland $122/year
    - Parallels desktop $100/year

# Minutes

- Primary goal of examining expenses around the project.
  - A salary for Randy to start, a second person has numerous benefits around redundancy and larger initiatives.
  - Test Runners
    - Mac Stadium = $100/mo but could get free as OSS project
    - No legit way through GitHub CI to install Docker Desktop on a test runner
    - So really, self-hosted runners for Windows &amp; Mac are what we need - and the ones we&#39;re currently using are getting very old. $2000/yr to replace self-hosted runners or growing them is good.
- Cary Gordon: coming from Drupal experience, watching Dries not taking a salary--that doesn&#39;t work here. I feel we really need to get more people involved with development. Potentially an issue with the language choice of Go with the fact that this is a PHP-facing community. So I think aiming for something like $100,000 will help us to grow in terms of those resources. Drupal phones home, but DDEV DOES have some telemetry.
- Brad: We have about 3200 weekly active users. I was thinking more like $150k budget.
  - 150k / 3200 = ~$45/year/weekly user
- CG: Do we \_charge\_ for it? Beg for it? Have an organization? Having an organization I think is really important.
- Randy: I am really a true fan of free open source. I would resist forcing people to pay for it. BUT, companies who are using it should just pony up. GitHub Sponsors for individuals… but it&#39;s not easy to get someone to pay for something that&#39;s free. But we could work toward our goals - companies that are using it know the value of it. They could pony up $100/mo or $1200/year… and we could get where we need to go with goals like that… _we&#39;re not in a hurry_.
- CG: First hire should be a Development officer (as in fundraising) - (after Randy… Randy is really the first hire).
- Andy Blum: What about DDEV &quot;PRO&quot; SAAS where users can have some kind of pro feature - especially team-based functionality that would be behind some sort of paywall.
- RF: MkDocs Material Theme does exactly that. They introduce features to the edge (subscribers). It does add friction to development. It&#39;s a legitimate thing to do.
- RF - we should set a goal for our budget for this year. I&#39;ve heard 100k and 150k. How do we get that? Not the thing today, but what should we say is the goal?
- AB: is there a way that people can donate today?
  - RF - yes there is a bank account. Cary tried to send money but it didn&#39;t work. Bank Account owned by the LocalDev Foundation, which does exist. We don&#39;t yet have the clarity to turn on Github Sponsors yet. That&#39;s easy for a lot of people.
- Brad has spent a lot of time on the DAO. Contracts and stuff are largely written. I think we could sell an NFT to raise the first year budget. Fruition would finance that and then transfer ownership of that to the DAO.
  - We have an active happy engaged group of 3k people, and we don&#39;t have a way for them to easily participate in the future. No skin in the game, but free is free. What if we get them to pony up a few hundred bucks, now there&#39;s skin in the game. Yeah NFTs are goofy, but then they help dictate where things go. It lets us do a bounty board for additional features. Would pull in people who are using it.
- CG: the bank we have right now is not great (something with Fidelity). It&#39;s not their job to take in all the money from all these sources.
- RF: I do think people will be excited to buy into a DAO. Worry more about the companies/agencies, who are really more our target audience. Is there also a &quot;money&quot; way to contribute and not just contribute to the DAO.
- Chris Wells: CiviCRM&#39;s &quot;Make it Happen&quot; model might be better than a &quot;DDEV PRO&quot; model where the features are behind it, but more like mini capital campaigns for features.
- RF: TAKEAWAYs - set a goal of 100-150k. Hire a second &quot;Developer&quot; or hire someone for &quot;Development&quot; (fundraising).
- CG: Meghan Sanicki may be able to help with Exec.Dir. stuff. Fine with the DAO but no standing to take a vote, so that&#39;s Randy&#39;s decision. Think we should set goal really high - more like 200k-250k.
- CW: buying things for DDEV that are specific motivates people. Most fundraising happens when the thermometer is right at the top.
- CG: but there&#39;s a strong tend toward subscription, because it lowers the weight of capital fundraising.
- AB: As far as from Lullabot I can&#39;t speak for them and I&#39;ve been with them \&lt; 1 year, so I don&#39;t know what they&#39;d say. But Lullabot sponsors conferences and initiatives like Discover Drupal so something like that is not out of the question for DDEV. They&#39;ve sponsored Lando in the past from marketing budget. Individuals may have tech stipends that they could put $10/mo into GitHub Sponsors. Somewhere around $40/mo I would start hitting friction.
- CG: Maybe a Drupalize.me connection might be beneficial as well. People just getting into Drupal could be prescribed an environment that &#39;just works&#39; - bring people in right at the beginning.
- CW: How does the DAO function with money vs. voting?
  - Brad Anderson: general tokens would be given for voting share, NFTs would be sold to fundraise. You&#39;re not necessarily buying voting rights.
- RF: we need to definitely fully understand the DAO if that&#39;s the way we go. If the DAO idea moves me away from what I love to do, that&#39;s no good.
- BA: the only governance items TBD are who will have the voting shares, and then the first proposal we do is to approve the budget. Brad is not worried about raising the first year of operating budget. Then, to Cary&#39;s point - how do we grow it.
- CG: A &quot;capital call&quot; is when you own shares of something and then they run out of money so they call you to say &quot;give us money now.&quot;
- AB: With regards to NFT sales and the DAO - what are your thoughts on the environmental impact of blockchain/crypto? That&#39;s one thing in this that&#39;s hairy for this community. Our contract specifically is very light. No miners, for example.
- CW: other concern about that is the ease of donation.
  - BA: we have to walk people through that. So we can&#39;t just accept cash? Go into it with a goal to raise through NFT&#39;s. We have a concierge who helps us do that. It is an opportunity to learn crypto.

Action items

- Turn on github sponsors
- Create a pathway for companies to contribute that can be easily accountable
- Get a bank account with a better provider
- Brad to share GitBook re: proposal outline.

**Comment by deviantintegral** [comment]
> No legit way through GitHub CI to install Docker Desktop on a test runner

@rfay has this been discussed with the Docker folks at all? I'm assuming this applies to any CI service, not just GitHub. If they've specifically said they won't support this use case, I'd be interested in a link to that discussion.

> As far as from Lullabot I can't speak for them and I've been with them &lt; 1 year, so I don't know what they'd say.

I can provide context for this:

- All team members have budget they can use for technology purchases, which is also fine to use for GitHub Sponsors and the like.
- As a company, we generally contribute to _organizations_ or contract _individuals_ for specific work.
- What I've been waiting for is a clear ask and plan from the LocalDev foundation to present to our leadership. Glad to chat further offline about this.

**Reply by rfay** [reply]
@deviantintegral see https://github.com/docker/roadmap/issues/80 :)  There is also https://github.com/docker/for-mac/issues/2359, where one person continues to figure out how to do it every time it gets broken. Neither has any response from the Docker folks. 

It's a good reason to move on to colima, which has official support in the soon-to-be-released v1.19.0. Thanks for your pioneering with that!

Happy to talk about all the latest updates (see https://github.com/drud/ddev/discussions/3650) - quite a lot of change!
